### PR TITLE
feat(gdk) added gdk-form

### PR DESCRIPTION
**Pull request by tigredonorte**

### CODE REVIEW

File: libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { GdkFormComponent } from './gdk-form.component';
+
+describe('GdkFormComponent', () => {
+  let component: GdkFormComponent;
+  let fixture: ComponentFixture<GdkFormComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [GdkFormComponent]
+    }).compileComponents();
```
Comment (oliveira-gust) [inline libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts]: Test is failling, probably due to missing imports in testing module

File: libs/gdk/src/lib/gdk-form/gdk-form.component.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+:host {
+    width: 100%;
+}
```
Comment (oliveira-gust) [inline libs/gdk/src/lib/gdk-form/gdk-form.component.scss]: Missing newline at end of file

File: libs/gdk/src/lib/gdk-form/gdk-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@Component({
+  selector: 'gdk-form',
+  templateUrl: './gdk-form.component.html',
+  styleUrls: ['./gdk-form.component.scss']
+})
+export class GdkFormComponent {
+  public form: FormGroup;
+  public model_: any[] = [];
+  public displayModel: any[] = [];
+  @Input() public set model(model: any[]) {
```
Comment (oliveira-gust) [inline libs/gdk/src/lib/gdk-form/gdk-form.component.ts]: Please add docs for this input

File: libs/gdk/src/lib/gdk-form/gdk-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    this.model_ = model;
+    this.form = this.initForm(model);
+    this.setDisplayModel(this.form.value);
+    this.form.valueChanges.subscribe(data => {
+      this.setDisplayModel(data);
+    });
+    this.getForm.emit(this.form);
+  }
+
+  @Output() getForm = new EventEmitter();
```
Comment (oliveira-gust) [inline libs/gdk/src/lib/gdk-form/gdk-form.component.ts]: Please add docs for this output

File: libs/gdk/src/lib/gdk-json-input/gdk-json-input.component.html [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<textarea #jsontext matInput
+    [value]="jsonString"
+    (change)="onChange($event)" 
+    (keyup)="onChange($event)"
+></textarea>
```
Comment (oliveira-gust) [inline libs/gdk/src/lib/gdk-json-input/gdk-json-input.component.html]: Missing newline at end of file

File: libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(GdkFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (oliveira-gust) [inline libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts]: _eye emoji_

File: libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import { GdkFormComponent } from './gdk-form.component';
+
+describe('GdkFormComponent', () => {
+  let component: GdkFormComponent;
+  let fixture: ComponentFixture<GdkFormComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [GdkModule],
+      // declarations: [GdkFormComponent]
```
Comment (guilhermejcgois) [inline libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts]: Commented code

File: package.json [hunk 8]
```diff
@@ -149,6 +149,7 @@
     "protractor": "~5.4.0",
     "replace": "1.1.1",
     "rimraf": "^2.6.3",
+    "rollup-plugin-node-builtins": "^2.1.2",
```
Comment (guilhermejcgois) [inline package.json]: Why exactly we need this dependency?

Comment (tigredonorte) [reply package.json]: I have no idea why it's here

File: libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts (hunk 7 above)
Comment (tigredonorte) [reply libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts]: ok

File: libs/gdk/src/lib/gdk-form/gdk-form.component.scss (hunk 2 above)
Comment (tigredonorte) [reply libs/gdk/src/lib/gdk-form/gdk-form.component.scss]: ok

File: libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts (hunk 1 above)
Comment (tigredonorte) [reply libs/gdk/src/lib/gdk-form/gdk-form.component.spec.ts]: ok

**Pull request closed without merging**